### ISSUE TITLE
Data corruption with NCCL Allgather when one NVLink is down

### ISSUE THREAD

**Issue by xw285cornell** [NVIDIA/nccl#596]
We're observing NCCL all gather produces corrupted data when one NVLink is down on a GPU, and we're doing all gather on multiple nodes. 

Hardware setup: 
DGX-1 like system, V100 SXM3 + NVSwitch + RDMA
two nodes

Software: NCCL 2.10.3

Repro: nccl-tests all gather

**Comment by jbachan** [comment]
Can you try the test under each of these environment settings: NCCL_PROTO=LL, NCCL_PROTO=LL128, NCCL_PROTO=SIMPLE. I have a hunch that the corruption only happens with LL128.

**Comment by xw285cornell** [comment]
Yes, I think we have some discussions with the NCCL team and we found LL128 is probably the root cause. cc @kwen2501

**Comment by christopherhesse** [comment]
This commit https://github.com/NVIDIA/nccl/commit/d427af5d94dc89925ecf366062736fb764389f59 says "Fix NVLink detection and avoid data corruption when some NVLinks are down." but all the commits are merged together, which changes actually fix that behavior?

**Comment by jbachan** [comment]
Here is the patch against 2.10.3. When applied to 2.11.4 I checked that it still produces a clean compilation and successfully runs a nccl test.
[github-2.10.3-1-p2p-detect.patch.txt](https://github.com/NVIDIA/nccl/files/8351952/github-2.10.3-1-p2p-detect.patch.txt)

**Comment by stephenmsachs** [comment]
### Issue Cleanup: Helping Us Focus on Current Challenges

  We're reviewing older issues to ensure we prioritize the most relevant and active ones. Since this issue hasn't seen updates in over 6 months, we'll be closing it for now.

  ,*This change helps us focus our efforts on addressing any current issues our users are facing.* If this issue still affects you, please don't hesitate to reopen it with a quick update (e.g., "Still relevant on [version=X]").
  Thanks for your understanding and for contributing to NCCL.
